service/src/main/java/uk/gov/hmcts/reform/fpl/enums/GeneratedOrderType.java

```diff
@@ -21,7 +21,11 @@ public static GeneratedOrderType fromType(String type) {
         type = type.replaceAll("(Final|Interim|\\(C21\\))", "").strip();
         type = type.toUpperCase();
         type = type.replace(" ", "_");
-        return GeneratedOrderType.valueOf(type);
+        try {
```

Comment:
could be
return ofNullable(EnumUtils.getEnum(GeneratedOrderType.class, type)).orElse(UPLOAD);
as this should not be based on exception

Reply:
the implementation of EnumUtils is actually the catch of the exception
    public static <E extends Enum<E>> E getEnum(Class<E> enumClass, String enumName) {
        if (enumName == null) {
            return null;
        } else {
            try {
                return Enum.valueOf(enumClass, enumName);
            } catch (IllegalArgumentException var3) {
                return null;
            }
        }
    }

Reply:
Wasn't aware of that class, if we want to use that method it could just be `EnumUtils.getEnum(GeneratedOrderType.class, type, UPLOAD)`

Reply:
reads better on your option anyway, I like it more. Pushed the change

Reply:
and we all learnt new thing 👍